homeassistant/components/syncthing/sensor.py

```diff
@@ -0,0 +1,224 @@
+"""Support for monitoring the Syncthing instance."""
+
+import logging
+
+import aiosyncthing
+
+from homeassistant.const import CONF_NAME
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import (
+    DOMAIN,
+    FOLDER_PAUSED_RECEIVED,
+    FOLDER_SENSOR_ALERT_ICON,
+    FOLDER_SENSOR_DEFAULT_ICON,
+    FOLDER_SENSOR_ICONS,
+    FOLDER_SUMMARY_RECEIVED,
+    SCAN_INTERVAL,
+    SERVER_AVAILABLE,
+    SERVER_UNAVAILABLE,
+    STATE_CHANGED_RECEIVED,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Syncthing sensors."""
+
+    name = config_entry.data[CONF_NAME]
+    syncthing = hass.data[DOMAIN][name]
+
+    try:
+        config = await syncthing.system.config()
+        version = await syncthing.system.version()
+        dev = [
+            FolderSensor(
+                hass,
+                syncthing,
+                name,
+                folder["id"],
+                folder["label"],
+                version["version"],
+            )
+            for folder in config["folders"]
+        ]
+
+        async_add_entities(dev)
+    except aiosyncthing.exceptions.SyncthingError as exception:
+        raise PlatformNotReady from exception
+
+
+class FolderSensor(Entity):
+    """A Syncthing folder sensor."""
+
+    def __init__(self, hass, syncthing, name, folder_id, folder_label, version):
+        """Initialize the sensor."""
+        self.hass = hass
+        self._syncthing = syncthing
+        self._name = name
+        self._folder_id = folder_id
+        self._folder_label = folder_label
+        self._state = None
+        self._unsub_timer = None
+        self._version = version
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._name} {self._folder_id} {self._folder_label}"
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the entity."""
+        return f"{DOMAIN}-{self._name}-{self._folder_id}"
```

Comment:
User configured names are frowned upon when used in unique_ids. Since you have a URL I was surprised you don't use this in the unique_id and in the config_entry's duplicate check.

Reply:
Sorry, I don't really get it, could you please give me a bit more details

Reply:
Name is just anything the user enters right? It isn't a semi-permanent attribute of the thing you are connecting to like its IP address/hostname/MAC address/etc?

Reply:
@joncar Yep, the name is set by the user during the integration setup. Would you recommend using the URL instead of the name? Does it need to be somehow escaped?

Reply:
Here's our docs about entity unique_id:
https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements

Reply:
@MartinHjelmare I could replace the name with provided by the syncthing server unique id everywhere in the integration, but this id is pretty long and has format of `xxxxxxx-xxxxxxx-xxxxxxx-xxxxxxx-xxxxxxx-xxxxxxx-xxxxxxx-xxxxxxx` where x is a digit or a letter in upper case. I don't think it will be handy to include this id in entity ids.

The other option would be using the first or last 7 symbols from the id, i believe they're random enough to fit the requirements, what do you think?

Reply:
The unique_id isn't user facing so it doesn't matter if it's long. The entity_id will use the name property as template so that should be usefully long.

Reply:
I'm aiming for completer emoving the name property. In case if a user adds multiple servers that share same directories, and uses the same name for all of them, it will cause entity id conflicts. I will use the first 7 bytes of the server id to form entity id. I believe this is the best balance of safety and usefulness. Users who want human readable server identifier in entity id are free to manually change it.

Reply:
We should use the name property. If you want to use the first 7 bytes of the server id for the name that's ok. It will be used for entity_id too.

Since we set a unique_id there won't be any conflicts. The unique_id keeps track of the entity.

Reply:
Sorry, I meant the name property in the config flow, of course the entity will have the name property.

Reply:
I'm assuming you mean the name config and form item. Yes, we don't want to ask the user for a name in the config flow.